### PR TITLE
tests - clean up deprecations

### PR DESCRIPTION
**Pull request by kapilt**
a few assorted cleanups on deprecations during test execution. after this the only deprecation warnings are from dependencies.

### CODE REVIEW

File: tools/c7n_azure/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-vcrpy==2.0.1
+vcrpy==2.1.0
```
Comment (stefangordon) [inline tools/c7n_azure/requirements.txt]: I had thought 2.1 wouldn't work but the build failures I'm seeing look unrelated (just a regex issue) - so hoping the issue was resolved in newer Azure SDK's.

**Comment by kapilt** [comment]
so roughly 65% of remaining deprecations are from azure key vault auth setup, that looks like its fixed upstream, but requires switching packages (they split out the package into three), all of which include the key vault extended oauth setup.

**Comment by stefangordon** [comment]
> so roughly 65% of remaining deprecations are from azure key vault auth setup, that looks like its fixed upstream, but requires switching packages (they split out the package into three), all of which include the key vault extended oauth setup.

@logachev fyi

**Comment by kapilt** [comment]
sounds good re parity on keyvault.

**Pull request merged**